partiql-ast/src/main/java/org/partiql/ast/dml/UpdateTargetStep.java

```diff
@@ -0,0 +1,115 @@
+package org.partiql.ast.dml;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.AstNode;
+import org.partiql.ast.AstVisitor;
+import org.partiql.ast.Identifier;
+import org.partiql.ast.expr.ExprLit;
+import org.partiql.value.PartiQL;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This references a column or column's nested properties. In SQL:1999, the EBNF rule is &lt;update target&gt;.
+ * This implementation differs from SQL by allowing for references to deeply nested data of varying types.
+ *
+ * @see SetClause
+ */
+public abstract class UpdateTargetStep extends AstNode {
+
+    /**
+     * This is a reference to a field of an array/struct using the bracket notation.
+     * @see UpdateTarget
+     * @see UpdateTargetStep
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static final class Element extends UpdateTargetStep {
+        // TODO: Should we explicitly have an ElementInt and ElementString? Especially once PartiQLValue is removed.
+
+        /**
+         * TODO
+         */
+        // TODO: Change this to a literal, not an ExprLit
```

Comment:
The updated literal modeling (https://github.com/partiql/partiql-lang-kotlin/pull/1650) has been merged in, so we can address this TODO. Think we'll have to otherwise build will fail.